source/ai/rl/BBF_agent/BBF.py

```
@@ -0,0 +1,442 @@
+        perception_modules = [self.model.encoder_cnn, self.model.transition]
+        actor_modules = [
+            self.model.prediction,
+            self.model.projection,
+            self.model.a,
+            self.model.v,
+        ]
+
+        params_wm = []
+        for module in perception_modules:
+            for param in module.parameters():
+                if param.requires_grad == True:  # They all require grad
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `param.requires_grad:` for truth checks

Replace with `param.requires_grad`

(E712)